lib/submit-analytics.js

```diff
@@ -33,50 +104,184 @@ function sha256(data) {
-        // Submit analytics after some time - we do this to give the application a chance to disable analytics.
+            // payloads for webhook and MixPanel differ slightly
+            const payload = {
+                'webHook': {
+                    'event': eventName,
+                    'properties': {
+                        'token': 'aab85907a13e1ff44a95be539d9942a9',
+                        'distinct_id': identifier,
+                        'Anonymized Machine Identifier': identifier,
+                        'Anonymized Application ID': sha256(__dirname),
+                        'Binding': 'javascript',
+                        'Target': environment,
+                        'Version': context.version,
+                        'Language': 'javascript',
+                        'OS Type': os.platform(),
+                        'OS Version': os.release(),
+                        'Node.js versions': process.versions
+                    }
+                },
+                'mixPanel': {
+                    'event': eventName,
+                    'properties': {
+                        'token': 'aab85907a13e1ff44a95be539d9942a9',
+                        'distinct_id': identifier,
+                        'Anonymized Machine Identifier': identifier,
+                        'Anonymized Application ID': sha256(__dirname),
+                        'Binding': environment,
+                        'Version': context.version,
+                        'Language': 'javascript',
+                        'OS Type': os.platform(),
+                        'OS Version': os.release(),
+                        'Node.js versions': process.versions
+                    }
+                }
+            };
+
+            resolve(payload);
+            return;
+        });
+    });
+}
+
+/**
+ * Generate and submit analytics data to servers.
+ * @param  {[webhookProdDetails|webhookStageDetails|mixpanelDetails]} requestFactory Submission URL generator
+ * @param  {Object} context contents of the app's package.json file
+ * @param  {string} eventName Custom tag that signifies what the app was doing when the analytics were submitted
+ * @returns {Promise} Promise that resolves when analytics data has been submitted
+ */
+async function submitAnalytics(requestFactory, context, eventName) {
+    return new Promise((resolve, reject) => {
         setTimeout(() => {
             if (isAnalyticsDisabled()) {
+                // do nothing
+                resolve();
                 return;
             }
 
-            machineId()
-                .catch() // Ignore errors
-                .then((identifier) => {
-                    if (!identifier) {
-                        identifier = sha256('unknown');
-                    }
+            fetchPlatformData(context, eventName)
+            .catch((e) => {
+                reject("Failed to fetch platform data  [Error: " + e + "]");
+                return;
```

Comment:
No need to return here and throw instances of `Error` instead of strings:

```suggestion
                const err = new Error(`Failed to fetch platform data: ${e.message}`);
                reject(err);
```

The code above assumes that lower levels are throwing instances of `Error` (to be able to read out the `message`);

I would however ask you to consider if wrapping this error is actually needed?